src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -47,8 +48,10 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
             {
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
+                // in case of SingleFile deployment, Assembly.Location is empty.
+                isSingleFile = string.IsNullOrEmpty(exeAssembly?.Location);
```

Comment:
`isSingleFile` will be true for the hosted case where `Assembly.GetEntryAssembly` returns null and that may not be single file. 

It is a bit confusing and it will lead to behavior change for the hosted case below. This behavior change may or may not be desirable.

Reply:
I think comparing with `<Unknown>` string was more robust than this check to fix the regression. Unless we have a better way to detect single file here, should I revert back to comparing with `<Unknown>` to avoid any confusion?

Reply:
Another approach is to make this change strict: ` exeAssembly != null && string.IsNullOrEmpty(exeAssembly!.Location)`.